Lib/test/test_class.py

```diff
@@ -503,6 +503,58 @@ def __eq__(self, other): return 1
 
         self.assertRaises(TypeError, hash, C2())
 
+    def testPredefinedAttrs(self):
+        o = object()
+
+        class Custom:
+            pass
+
+        c = Custom()
+
+        methods = (
+            "__new__", "__init__",
+            "__repr__", "__str__", "__format__",
+            "__eq__", "__ne__", "__lt__", "__le__", "__gt__", "__ge__",
+            "__hash__",
+            "__getattribute__", "__setattr__", "__delattr__", "__dir__",
+        )
+        for name in methods:
+            with self.subTest(name):
+                self.assertTrue(callable(getattr(object, name)))
+                self.assertTrue(callable(getattr(o, name)))
+                self.assertTrue(callable(getattr(Custom, name)))
+                self.assertTrue(callable(getattr(c, name)))
+
+        not_defined = [
+            "__bool__", "__bytes__", "__del__", "__getattr__", "__len__",
+            "__get__", "__set__", "__delete__", "__objclass__",
+            "__len__", "__length_hint__", "__iter__", "__reversed__",
+            "__getitem__", "__setitem__", "__delitem__",
+            "__contains__", "__missing__",
+            "__divmod__", "__rdivmod__",
+            "__neg__", "__pos__", "__abs__", "__invert__",
+            "__complex__", "__int__", "__float__", "__round__", "__index__",
+            "__enter__", "__exit__",
+            "__await__", "__aiter__", "__anext__", "__aenter__", "__aexit__",
+        ]
+        augment = (
+            "add", "sub",
+            "mul", "matmul", "truediv", "floordiv", "mod", "pow",
+            "lshift", "rshift", "and", "xor", "or",
```

Comment:
It's interesting. Seems that `hasattr(object, "__or__")` and `hasattr(object, "__ror__")` is true. 
Maybe it's introduced after accepting [PEP 604](https://peps.python.org/pep-0604/).

Reply:
It seemed strange to me at first, and I couldn't figure out why. 

```
commit 1b4552c5e8e925f24c15f707050f22c977171125 (HEAD -> main)
Author: Maggie Moss <MaggieMoss@users.noreply.github.com>
Date:   Wed Sep 9 13:23:24 2020 -0700

    bpo-41428: Implementation for PEP 604 (GH-21515)
    
    See https://www.python.org/dev/peps/pep-0604/ for more information.
    
    Co-authored-by: Pablo Galindo <pablogsal@gmail.com>

commit fa8c9e70104b0aef966a518eb3a80a4881906ae0
Author: Jakub Kulík <Kulikjak@gmail.com>
Date:   Wed Sep 9 21:29:42 2020 +0200

    bpo-41687: Fix error handling in Solaris sendfile implementation (GH-22128)
    
    
    I just realized that my recent PR with sendfile on Solaris ([PR 22040](https://github.com/python/cpython/pull/22040)) has broken error handling.
    
    Sorry for that, this simple followup fixes that.
    
    Automerge-Triggered-By: @1st1
```

The change has been merged with [1b4552c5e8e925f24c15f707050f22c977171125](https://github.com/python/cpython/pull/21515). I took a Python built with the previous commit and I didn't see this behavior. Yes, we introduced this behavior after accepting [PEP 604](https://peps.python.org/pep-0604/). Good catch! @Eclips4